docs/examples/example_03_linux_socketcan_transport.cpp

```diff
@@ -0,0 +1,319 @@
+/// @file
+/// Example of creating a libcyphal node in your project using posix SOCKETCAN media and CAN transport.
+///
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+///
+
+#include "platform/posix/posix_single_threaded_executor.hpp"
+#include "platform/tracking_memory_resource.hpp"
+
+#ifdef __linux__
+#    include "platform/linux/can/can_media.hpp"
+#endif
+
+#include <cassert>  // NOLINT for NUNAVUT_ASSERT
+#include <nunavut/support/serialization.hpp>
+#include <uavcan/node/Health_1_0.hpp>
+#include <uavcan/node/Heartbeat_1_0.hpp>
+#include <uavcan/node/Mode_1_0.hpp>
+
+#include <cetl/pf17/cetlpf.hpp>
+#include <cetl/pf20/cetlpf.hpp>
+#include <libcyphal/executor.hpp>
+#include <libcyphal/transport/errors.hpp>
+#include <libcyphal/transport/msg_sessions.hpp>
+#include <libcyphal/transport/transport.hpp>
+#include <libcyphal/transport/types.hpp>
+#include <libcyphal/transport/can/media.hpp>
+#include <libcyphal/transport/can/can_transport.hpp>
+#include <libcyphal/transport/can/can_transport_impl.hpp>
+#include <libcyphal/types.hpp>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <array>
+#include <chrono>
+#include <cstddef>
+#include <cstdint>
+#include <cstdlib>
+#include <iomanip>
+#include <iostream>
+#include <locale>
+#include <string>
+#include <utility>
+
+namespace
+{
+
+using namespace libcyphal::transport;       // NOLINT This our main concern here in this test.
+using namespace libcyphal::transport::can;  // NOLINT This our main concern here in this test.
+
+using Duration            = libcyphal::Duration;
+using TimePoint           = libcyphal::TimePoint;
+using Callback            = libcyphal::IExecutor::Callback::Any;
+using CanTransportPtr     = libcyphal::UniquePtr<ICanTransport>;
+using MessageRxSessionPtr = libcyphal::UniquePtr<IMessageRxSession>;
+using MessageTxSessionPtr = libcyphal::UniquePtr<IMessageTxSession>;
+
+// https://github.com/llvm/llvm-project/issues/53444
+// NOLINTBEGIN(misc-unused-using-decls, misc-include-cleaner)
+using std::literals::chrono_literals::operator""s;
+using std::literals::chrono_literals::operator""ms;
+// NOLINTEND(misc-unused-using-decls, misc-include-cleaner)
+
+using testing::_;
+using testing::Eq;
+using testing::IsEmpty;
+using testing::VariantWith;
+
+// NOLINTBEGIN(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+
+class Example_03_LinuxSocketCanTransport : public testing::Test
+{
+protected:
+    void SetUp() override
+    {
+        std::cerr.imbue(std::locale("en_US.UTF-8"));
```

Comment:
These throw on osx. I know this is a Linux example but we're doing the same thing over on some posix examples that otherwise run fine on mac. Can we just omit the imbue calls?

Reply:
I did it just for nice formatting of integers printed to console. ok, I'll remove them.